cmd/juju/cloud/updatecredential.go

```diff
@@ -101,57 +136,244 @@ func (c *updateCredentialCommand) getAPI() (credentialAPI, error) {
 
 // Run implements Command.Run
 func (c *updateCredentialCommand) Run(ctx *cmd.Context) error {
-	cloud, err := common.CloudByName(c.cloud)
-	if errors.IsNotFound(err) {
-		ctx.Infof("Cloud %q not found", c.cloud)
-		return nil
-	} else if err != nil {
-		return err
+	// If no file and no cloud is provided, switch to interactive mode.
+	if c.CredentialsFile == "" && c.cloud == "" {
+		// TODO (anastasiamac 2019-03-22) interactive mode
+		return errors.NotSupportedf("`update-credential` in interactive mode")
 	}
-	getCredentialsParams := modelcmd.GetCredentialsParams{
-		Cloud:          *cloud,
-		CredentialName: c.credential,
+	var credentials map[string]jujucloud.CloudCredential
+	var err error
+	if c.CredentialsFile != "" {
+		credentials, err = credentialsFromFile(c.CredentialsFile, c.cloud, c.credential)
+		if err != nil {
+			logger.Errorf("%v", err)
```

Comment:
I think in the original code we were more pedantic about what gets to the console and what goes into client side log (if one exists). 
But since you and I talked and both agreed that it may not be xtra helpful at times, I have simplified this code and am returning the original error rather than logging it and returning ErrSilent instead.